File: dataforge-plots/plots-jfc/src/main/java/hep/dataforge/plots/tests/PlotContainerTest.java
/*
 * Copyright 2015 Alexander Nozik.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package hep.dataforge.plots.tests;

import hep.dataforge.meta.MetaBuilder;
import hep.dataforge.plots.data.PlottableData;
import hep.dataforge.plots.data.PlottableXYFunction;
import hep.dataforge.plots.fx.FXPlotUtils;
import hep.dataforge.plots.fx.PlotContainer;
import hep.dataforge.plots.jfreechart.JFreeChartFrame;
import hep.dataforge.tables.*;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Function;

/**
 *
 * @author Alexander Nozik
 */
public class PlotContainerTest {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        PlotContainer container = FXPlotUtils.displayContainer("My test container", 800, 600);

        JFreeChartFrame frame = new JFreeChartFrame();

        container.setPlot(frame);

        Function<Double,Double> func = (x1) -> x1 * x1;

        PlottableXYFunction funcPlot = PlottableXYFunction.plotFunction("func", func, 0.1, 4, 200);

        frame.add(funcPlot);

        String[] names = {"myX", "myY", "myXErr", "myYErr"};

        List<DataPoint> data = new ArrayList<>();
        data.add(new MapPoint(names, 0.5d, 0.2, 0.1, 0.1));
        data.add(new MapPoint(names, 1d, 1d, 0.2, 0.5));
        data.add(new MapPoint(names, 3d, 7d, 0, 0.5));
        Table ds = new ListTable(data);

        PlottableData dataPlot = PlottableData.plot("dataPlot", new XYAdapter("myX", "myXErr", "myY", "myYErr"), ds);

        frame.getConfig().setNode(new MetaBuilder("yAxis").putValue("type", "log"));

        frame.add(dataPlot);
    }

}

File: dataforge-plots/plots-demo/src/main/kotlin/hep/dataforge/plots/demo/PlotContainerTest.kt
/*
 * Copyright 2015 Alexander Nozik.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package hep.dataforge.plots.demo

import hep.dataforge.kodex.fx.plots.PlotManager
import hep.dataforge.meta.MetaBuilder
import hep.dataforge.plots.data.DataPlot
import hep.dataforge.plots.data.XYFunctionPlot
import hep.dataforge.tables.ListTable
import hep.dataforge.tables.ValueMap
import hep.dataforge.tables.XYAdapter
import hep.dataforge.values.Values
import java.util.*


/**
 * @param args the command line arguments
 */

fun main(args: Array<String>) {

    val func = { x: Double -> Math.pow(x, 2.0) }

    val funcPlot = XYFunctionPlot.plotFunction("func", func, 0.1, 4.0, 200)


    val names = arrayOf("myX", "myY", "myXErr", "myYErr")

    val data = ArrayList<Values>()
    data.add(ValueMap.of(names, 0.5, 0.2, 0.1, 0.1))
    data.add(ValueMap.of(names, 1.0, 1.0, 0.2, 0.5))
    data.add(ValueMap.of(names, 3.0, 7.0, 0, 0.5))
    val ds = ListTable(data)

    val dataPlot = DataPlot.plot("dataPlot", XYAdapter("myX", "myXErr", "myY", "myYErr"), ds)


    val manager = PlotManager();
    manager.startGlobal();

    manager.display("test", "testLog") {
        add(funcPlot)
        config.setNode(MetaBuilder("yAxis").putValue("type", "log"))
        add(dataPlot)
    }
    manager.display("test", "test") {
        add(funcPlot)
        add(dataPlot)
    }
    manager.display("test1") {
        add(funcPlot)
    }

}


